src/utils/FileManager.ts

```diff
@@ -0,0 +1,67 @@
+import { promisify } from 'util';
+import {
+  access as accessCb,
+  constants,
+  mkdir as mkdirCb,
+  unlink as unlinkCb,
+  writeFile as writeFileCb
+} from 'fs';
+import { Logger } from './Logger';
+
+export class FileManager {
+  private static _instance: FileManager;
+
+  static get Instance(): FileManager {
+    if (!this._instance) {
+      this._instance = new FileManager();
+    }
+
+    return this._instance;
+  }
+
+  public readonly _access = promisify(accessCb);
+  public readonly _unlink = promisify(unlinkCb);
+  public readonly _writeFile = promisify(writeFileCb);
+  public readonly _mkdir = promisify(mkdirCb);
+
+  public async writeFile(path: string, data: string): Promise<void> {
+    try {
+      await this.removeFile(path);
+      await this._writeFile(path, data);
+    } catch (e) {
+      Logger.Instance.err(e);
+    }
+  }
+
+  public async createIfIsNotExist(path: string): Promise<void> {
```

Comment:
`create` should be enough (_if not exists_ is an implementation detail).

Reply:
I prefer self-documenting source code instead jsdoc)

Reply:
Still, API shouldn't leak implementation details.

Reply:
I agree with you regarding public API

Reply:
Anyway, we should not confuse abstraction and naming convention.

Reply:
Moved to #28